pkg/controller/servicing/lister.go

```diff
@@ -194,14 +237,24 @@ func (d *NodeLister) Replace() []*core_v1.Node {
 
 	klusterVersion, err := version.ParseSemantic(d.Kluster.Status.ApiserverVersion)
 	if err != nil {
-		d.Logger.Log(
-			"msg", "Couldn't parse Kluster version. Skipping node upgrade.",
-			"err", err,
-		)
-		return found
+		klusterVersion = nil
 	}
 
 	for _, node := range upgradable {
+		if util.IsCoreOSNode(node) && util.IsFlatcarNodePool(nodeNameToPool[node.GetName()]) {
```

Comment:
This confused me also. I suppose, the idea is that we somehow flip the `image` in a nodePool from CoreOS to Flatcar and this marks the node as `upgradable`?

Reply:
If current node image differs from nodepool it will be marked as upgradable (and eventually replaced).